src/scaffoldmaker/utils/cylindermesh.py

```diff
@@ -786,13 +789,25 @@ def createEllipsePerimeter(centre, majorAxis, minorAxis, elementsCountAround, he
     unitMajorAxis = vector.normalise(majorAxis)
     unitMinorAxis = vector.normalise(minorAxis)
     useHeight = min(max(0.0, height), 2.0 * magMajorAxis)
-    totalRadians = geometry.getEllipseRadiansToX(magMajorAxis, 0.0, magMajorAxis - useHeight,
-                                                 initialTheta=0.5 * math.pi * useHeight / magMajorAxis)
+    # check if it is a circle
```

Comment:
I'm not happy with having circle fixes here.
I'd prefer as a first fix that the ellipse functions do exact circle calculations if major and minor axes are equal (to a tolerance of total value).
I also request you take the time to make the general ellipse functions near exact precision as it's now pretty important, I can't spare the time, plus I think you're well capable. You'll have to do some brainstorming and come back to me with plans and algorithms, but I have some suggestions:
1. make the ellipse perimeter function exact to a specified precision using iteration with series sums: have a think or look up algorithms such as one of these https://www.mathsisfun.com/geometry/ellipse-perimeter.html
2. the function for advancing the angle around by an arc length around an ellipse accrues error. See about making it work to specified precision like the above. I note that many of the use cases could be changed to instead advance a fraction of N equal sized segments around the ellipse and restore precision on each quarter ellipse, or we may pass in the pre-calculated exact perimeter and current distance around to continuously improve the regular function.

Reply:
Changed the code to be more precise.